slice/test/e2e/jobset_test.go

```diff
@@ -417,5 +422,121 @@ var _ = ginkgo.Describe("JobSet", func() {
+				Obj()
+
+			ginkgo.By("Creating a JobSet", func() {
+				utils.MustCreate(ctx, k8sClient, jobSet)
+			})
+
+			createdWorkload := &kueue.Workload{}
+			wlKey := types.NamespacedName{
+				Name:      jobsetcontroller.GetWorkloadNameForJobSet(jobSet.Name, jobSet.UID),
+				Namespace: ns.Name,
+			}
+
+			createdSlice := &slice.Slice{}
+			sliceKey := types.NamespacedName{
+				Name:      wlKey.Name,
+				Namespace: wlKey.Namespace,
+			}
+
+			ginkgo.By("Checking that Slice is created", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, sliceKey, createdSlice)).To(gomega.Succeed())
+				}, utils.Timeout, utils.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Adding Ready condition", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, sliceKey, createdSlice)).To(gomega.Succeed())
+					meta.SetStatusCondition(&createdSlice.Status.Conditions, metav1.Condition{
+						Type:    string(slice.Forming),
+						Status:  metav1.ConditionFalse,
+						Reason:  "Test",
+						Message: "Test",
+					})
+					meta.SetStatusCondition(&createdSlice.Status.Conditions, metav1.Condition{
+						Type:    string(slice.Ready),
+						Status:  metav1.ConditionTrue,
+						Reason:  "Test",
+						Message: "Test",
+					})
+					g.Expect(k8sClient.Status().Update(ctx, createdSlice)).To(gomega.Succeed())
+				}, utils.Timeout, utils.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Checking that the Workload is admitted", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, wlKey, createdWorkload)).Should(gomega.Succeed())
+					g.Expect(workload.IsAdmitted(createdWorkload)).Should(gomega.BeTrue())
+				}, utils.LongTimeout, utils.Timeout).Should(gomega.Succeed())
+			})
+
+			pods := &corev1.PodList{}
+
+			ginkgo.By("Checking that all pods are running", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.List(ctx, pods, client.InNamespace(ns.Name))).To(gomega.Succeed())
+					g.Expect(pods.Items).Should(gomega.HaveLen(1))
+					for _, pod := range pods.Items {
+						g.Expect(pod.Status.Phase).To(gomega.Equal(corev1.PodRunning))
+					}
+				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Deactivating the Workload", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, wlKey, createdWorkload)).Should(gomega.Succeed())
+					createdWorkload.Spec.Active = ptr.To(false)
+					g.Expect(k8sClient.Update(ctx, createdWorkload)).Should(gomega.Succeed())
+				}, utils.Timeout, utils.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Checking that Slice is still exists and waiting for Pods termination", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, sliceKey, createdSlice)).To(gomega.Succeed())
+				}, utils.ConsistentDuration, utils.ShortInterval).Should(gomega.Succeed())
+			})
```

Comment:
This looks like a flake risk? Do we have a way of preventing the pod from terminating by an action in the test?

Reply:
> Do we have a way of preventing the pod from terminating by an action in the test?

I couldn't find any better solution. We still need some form of graceful shutdown to test this.

Reply:
What about setting the LifecyclePreStopSleepSeconds to something high (1 minute), and then manually manipulating/killing the pod?

Reply:
Done